octopus_deploy/datadog_checks/octopus_deploy/check.py

```diff
@@ -49,17 +56,58 @@ def _initialize_caches(self):
         for _, _, project_group, project_group_config in self.project_groups():
             self._initialize_projects(project_group, project_group_config)
 
+    def _update_completed_times(self):
+        previous_to_time = self._to_completed_time
```

Comment:
Calling the `get_current_datetime()` method can be slow in terms of performance.
How about creating a `self._check_time` member with the value of `get_current_datetime()` and being able to use it at different points in the integration code?

Reply:
Updated to only calculate it once!